### PR TITLE
[cleanup] RulesDeploy overhall

### PR DESCRIPTION
**Pull request by bkendall**
### Description

Some cleanup of `RulesDeploy`:

- renamed the file to match the rest of the codebase
- migrated the `firestore` and `storage` deploy folders to Typescript
- added tests for `src/rulesDeploy.ts`

### Scenarios Tested

- [x] deploy Firestore rules
- [x] deploy Storage rules

### TODOs

- [x] complete tests
- [x] functionally test these behaviors

### CODE REVIEW

**Comment by samtstern** [comment]
@bkendall let me know when this is not a draft.

File: src/rulesDeploy.ts [hunk 1]
```diff
@@ -175,15 +175,19 @@ export class RulesDeploy {
...
           const entriesToDelete = unreleased.reverse().slice(0, RULESETS_TO_GC);
-          for (const entry of entriesToDelete) {
-            await gcp.rules.deleteRuleset(this.options.project, gcp.rules.getRulesetId(entry));
-            logger.debug(`[rules] Deleted ${entry.name}`);
-          }
+          await Promise.all(
+            entriesToDelete.map(async (entry) => {
+              await gcp.rules.deleteRuleset(this.options.project, gcp.rules.getRulesetId(entry));
+              logger.debug(`[rules] Deleted ${entry.name}`);
+            })
```
Comment (ryanpbrewster) [inline src/rulesDeploy.ts]: please revert this change, these deletes should happen in series to avoid running into backend quota limits

((can you add a comment to that effect? this was a very reasonable optimization to make, we should explain why we didn't take it))

Comment (bkendall) [reply src/rulesDeploy.ts]: Will do. Good eye.

File: src/deploy/firestore/release.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { get } from "lodash";
```
Comment (ryanpbrewster) [inline src/deploy/firestore/release.ts]: mild preference for `import * as _ from "lodash"`

Comment (bkendall) [reply src/deploy/firestore/release.ts]: sure.
